src/Symfony/Contracts/EventDispatcher/EventDispatcherInterface.php

```diff
@@ -13,46 +13,19 @@
 
 use Psr\EventDispatcher\EventDispatcherInterface as PsrEventDispatcherInterface;
 
-if (interface_exists(PsrEventDispatcherInterface::class)) {
-    /**
-     * Allows providing hooks on domain-specific lifecycles by dispatching events.
-     */
-    interface EventDispatcherInterface extends PsrEventDispatcherInterface
-    {
-        /**
-         * Dispatches an event to all registered listeners.
-         *
-         * For BC with Symfony 4, the $eventName argument is not declared explicitly on the
-         * signature of the method. Implementations that are not bound by this BC constraint
-         * MUST declare it explicitly, as allowed by PHP.
-         *
-         * @param object      $event     The event to pass to the event handlers/listeners
-         * @param string|null $eventName The name of the event to dispatch. If not supplied,
-         *                               the class of $event should be used instead.
-         *
-         * @return object The passed $event MUST be returned
-         */
-        public function dispatch($event/*, string $eventName = null*/);
-    }
-} else {
+/**
+ * Allows providing hooks on domain-specific lifecycles by dispatching events.
+ */
+interface EventDispatcherInterface extends PsrEventDispatcherInterface
+{
     /**
-     * Allows providing hooks on domain-specific lifecycles by dispatching events.
+     * Dispatches an event to all registered listeners.
+     *
+     * @param object      $event     The event to pass to the event handlers/listeners
+     * @param string|null $eventName The name of the event to dispatch. If not supplied,
+     *                               the class of $event should be used instead.
+     *
+     * @return object The passed $event MUST be returned
      */
-    interface EventDispatcherInterface
-    {
-        /**
-         * Dispatches an event to all registered listeners.
-         *
-         * For BC with Symfony 4, the $eventName argument is not declared explicitly on the
-         * signature of the method. Implementations that are not bound by this BC constraint
-         * MUST declare it explicitly, as allowed by PHP.
-         *
-         * @param object      $event     The event to pass to the event handlers/listeners
-         * @param string|null $eventName The name of the event to dispatch. If not supplied,
-         *                               the class of $event should be used instead.
-         *
-         * @return object The passed $event MUST be returned
-         */
-        public function dispatch($event/*, string $eventName = null*/);
-    }
+    public function dispatch(object $event, string $eventName = null): object;
```

Comment:
This signature change causes a Fatal error.

```
Fatal error: Declaration of Symfony\Contracts\EventDispatcher\EventDispatcherInterface::dispatch(Symfony\Contracts\EventDispatcher\object $event, ?string $eventName = NULL): Symfony\Contracts\EventDispatcher\object must be compatible with Psr\EventDispatcher\EventDispatcherInterface::dispatch(Psr\EventDispatcher\object $event) in
```

I'm afraid there is a signature mismatch. 
https://github.com/php-fig/event-dispatcher/blob/cfea31e2e2abbe98159e825ce7d5040a39fba5f0/src/EventDispatcherInterface.php#L20

Reply:
Oh sorry, I found out it does not happen in php 7.2.

Reply:
Exactly. Neither v2 of the contracts nor the PSR interface are compatible with php 7.1.

Reply:
@DQNEO Make sure to run `composer update` in an environment that uses the same PHP your code is executed with (or configure the target PHP version using the platform config). Otherwise you risk ending up with packages being installed that are not compatible with your runtime environment.

Reply:
It happened when I used php-cs-fixer with php7.1. 
So the cause is in that package.
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.16/composer.json#L25
Thank you for your advise !